### ISSUE TITLE
Determine provider discrepancies on `get_response_json` try/catch

### ISSUE THREAD

**Issue by AetherUnbound** [WordPress/openverse-catalog#703]
## Description

As part of #555 and #645, we unified the provider ingester script query mechanisms and error handling. In the process however, it looks like we may have introduced error handling/swallowing where there previously was none:

https://github.com/WordPress/openverse-catalog/blob/eec83ac6970dd73cb5e5efec140c7a0f540c74f1/openverse_catalog/dags/providers/provider_api_scripts/provider_data_ingester.py#L296-L306

As written, this will catch every failure that could reasonably happen from `get_response_json`, swallow it, and continue execution. For **most** cases, this is neither ideal or appropriate. However for some cases (e.g. [Brooklyn Museum](https://github.com/WordPress/openverse-catalog/pull/645/files#diff-7c4ffca320da8f2f4e9382b7a90945e5cad454dd4c633168ba9f79e4312e7da3)), this was the intended behavior. It seems that this probably _shouldn't_ be the case though, since #650 gave us the capacity to skip ingestion errors that occur within a batch. We can rely on this method for catching errors rather than adding an explicit try/catch here.

This will involve altering tests, as some providers expect errors to result in continued execution.

**In summary**, we want to change this to halt execution on errors that occur here, since we have higher-order mechanisms for skipping/handling errors in processing.

## Resolution

- [ ] 🙋 I would be interested in resolving this bug.
